test/thing-unit-tests.js

```diff
@@ -69,6 +69,80 @@ describe( "thing shadow class unit tests", function() {
       });
    });
 
+   describe( "register a thing shadow name", function() {
+//
+// Verify that the thing shadow invokes the register callback when subscription to all
+// topics are finished. The callback is invoked based on the callback from the mqtt library.
+//
+
+      var thingShadowsConfig = {
+         keyPath: 'test/data/private.pem.key',
+         certPath: 'test/data/certificate.pem.crt',
+         caPath: 'test/data/root-CA.crt',
+         clientId: 'dummy-client-1',
+         region: 'us-east-1'
+      };
+
+      it("should trigger error when a subscription fails", function () {
+
+        // Need local stub to trigger error, so just copy-paste the code
+        var localMockMqtt;
+        var fakeConnect = function (options) {
```

Comment:
Needed the possibility to force an error to get the test working. So reused the code from the beforeEach.